### PR TITLE
feat(TBD-4197): set hive query name for tracking

### PR DESCRIPTION
**Pull request by yu-talend**
For tHiveRow, every time when we submit a query to HiveServer2, setting the application name (for Mapreduce Engine) or DAG name (for Tez engine) as : talend_{job name}_{timestamp}, and show it in console under info level logs. So that users can track this job on Yarn RM UI or Tez UI.

Associated modification for tbd-studio-se: https://github.com/Talend/tbd-studio-se/pull/477

Jira link: https://jira.talendforge.org/browse/TBD-4197

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet [hunk 1]
```diff
@@ -186,7 +191,10 @@ if(hiveDistrib.isExecutedThroughWebHCat()) {
 	}
 	%>
 	try {
-	<%		
+		final String queryIdentifier = "talend_" + jobName + "_<%=currentTime%>";
```
Comment (bchen-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet]: the time is the code generated time, not the job running time, it's ok?

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet [hunk 2]
```diff
@@ -186,7 +191,10 @@ if(hiveDistrib.isExecutedThroughWebHCat()) {
...
+		stmt_<%=cid %>.execute("set mapred.job.name=" + queryIdentifier);
+		stmt_<%=cid %>.execute("set mapred.job.name=" + queryIdentifier);
```
Comment (bchen-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet]: duplicate code with line:195?

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet [hunk 3]
```diff
@@ -229,16 +229,14 @@ imports="
...
 			java.sql.PreparedStatement pstmt_<%=cid %> = conn_<%=cid%>.prepareStatement(<%=dbquery%>);
-<%
-		} else {
-%>
-			java.sql.Statement stmt_<%=cid %> = conn_<%=cid %>.createStatement();
-<%
+		<%
 		}
+		%>
+		java.sql.Statement stmt_<%=cid %> = conn_<%=cid %>.createStatement();
```
Comment (bchen-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet]: should close in the end.javajet

**Comment by bchen-talend** [comment]
the change for test/plugins/org.talend.designer.scd.test/targets/.gitignore is needed?

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet [hunk 4]
```diff
@@ -229,16 +229,14 @@ imports="
 
 	if(!hiveDistrib.useCloudLauncher()) {
 		if (usePrepareStatement) {
-%>
+		%>
 			java.sql.PreparedStatement pstmt_<%=cid %> = conn_<%=cid%>.prepareStatement(<%=dbquery%>);
-<%
-		} else {
```
Comment (amarouni) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet]: Why you're removing this ? Why do you need this change ?

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet (hunk 1 above)
Comment (amarouni) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet]: As @bchen-talend said there's `jobTime` or something like this that you should use.

File: test/plugins/org.talend.designer.scd.test/targets/.gitignore [hunk 5]
```diff
@@ -1 +0,0 @@
-/org/
```
Comment (amarouni) [inline test/plugins/org.talend.designer.scd.test/targets/.gitignore]: Is this intentional ?

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet (hunk 4 above)
Comment (yu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet]: In fact this is to make it also work for `PrepareStatement`, cause we cannot execute a customer query for `PrepareStatement` and for it the "Statement" is already set we can just execute, so I just created a `HiveStatement` in this case to execute the set query name request.

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet [hunk 6]
```diff
@@ -186,7 +186,12 @@ if(hiveDistrib.isExecutedThroughWebHCat()) {
 	}
 	%>
 	try {
-	<%		
+		final String queryIdentifier = "talend_" + jobName + "_" + new Date(startTime);
```
Comment (bchen-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet]: @amarouni @yu-talend 
one question about the name,
so we also need to add component name, as one talend job may contains multiple hive relate components.
that is copied from spark streaming job, but with `cid` in the middle
```
java.text.DateFormat outdfm = new java.text.SimpleDateFormat("yyyy_MM_dd_HH_mm_ss");
        String appName = projectName + "_" + jobName + "_" + jobVersion + "_" + cid + "_" + outdfm.format(new java.util.Date());
```

Comment (amarouni) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet]: Yes, good point @bchen-talend. @yu-talend, in a job you might have multiple Hive components that can run jobs on the cluster so we have to add the component id `cid` to the unique app name. Something like `projectName + "_" + jobName + "_" + jobVersion + "_" + cid + "_" + outdfm.format(new java.util.Date());`. The code can be found in spark_footer.javajet

Comment (yu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet]: +1

File: main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Input/AbstractDBInputBegin.javajet [hunk 7]
```diff
@@ -78,23 +78,26 @@ imports="
...
-
+            %>
 		    java.sql.ResultSet rs_<%=cid %> = null;
-		try{
-		    rs_<%=cid %> = stmt_<%=cid %>.executeQuery(dbquery_<%=cid%>);
-		    java.sql.ResultSetMetaData rsmd_<%=cid %> = rs_<%=cid %>.getMetaData();
-		    int colQtyInRs_<%=cid%> = rsmd_<%=cid %>.getColumnCount();
+
+		    try{
+		    	if (stmt_<%=cid %> instanceof org.apache.hive.jdbc.HiveStatement) {
```
Comment (bchen-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Input/AbstractDBInputBegin.javajet]: AbstractDBInputBegin.javajet is abstract javajet, could used by different db components other than hive, so check org.apache.hive.jdbc.HiveStatement on runtime will cause class not found issue.

please look at org.talend.designer.components.localprovider/components/templates/DB/Input/HelpClass.javajet
DefaultDBInputUtil#createStatement
in HiveInput_begin.javajet, it define DBInputBeginUtil which extends DefaultDBInputUtil

**Comment by amarouni** [comment]
@yu-talend Are there any Hive components left in tdi-se that we need to modify ?

**Comment by yu-talend** [comment]
@amarouni normally it's all done, we have 4 components in tdi-se, `tHiveRow` is done, and the 3 others `tHiveConnection`, `tHiveClose`, and `tHiveIn` which are not concerned to execute query to HiveServer.

**Pull request merged**